pkg/js/transformer/jsontodsl.ts

```diff
@@ -390,3 +390,38 @@ function sortByModule(aName: string, bName: string, aMeta?: Metadata, bMeta?: Me
   // If the module name and file name are the same then sort based on name
   return aName.localeCompare(bName);
 }
+
+/* This function gets the modules from the JSON model and
+ * returns them as an alphabetically sorted array
+ */
+export function getModulesFromJSON(model: Omit<AuthorizationModel, "id">): string[] {
+  const schemaVersion = model?.schema_version || "1.1";
+
+  if (schemaVersion !== "1.2") {
+    throw new UnsupportedModularModules(schemaVersion);
+  }
+
+  const isModularModel = model.type_definitions?.some((typeDef) => typeDef.metadata?.module);
+
+  if (!isModularModel) {
+    return [];
+  }
+
+  const modules = model.type_definitions?.reduce((acc: (string | undefined)[], typeDef) => {
```

Comment:
Instead of using reduce, use a map instead.

```ts
const modulesMap = {};
model.type_definitions?.forEach(typeDef => {
  if (typeDef.metadata?.module) {
    modulesMap[typeDef.metadata.module] = true;
  }

  for (const relationMeta in typeDef?.metadata?.relations) {
    if (relationMeta?.module) {
      modulesMap[relationMeta.module] = true;
    }
  }
})

return Object.keys(modulesMap).sort()
```

With this you don't need reduce or filter and it will be more efficient, as you are looping only once.

The logic above re checking if modular can be removed too.